app/uploaders/file_upload.rb

```diff
@@ -13,11 +13,16 @@ def initialize(**args)
 
   def start!(file, trace: nil)
     # run the shrine upload process.
-    @attacher.assign(file)
+    if file.is_a? Shrine::UploadedFile
+      @attacher.set(file)
+    else
+      @attacher.assign(file)
+    end
     raise ArgumentError, @attacher.errors.join(',') unless @attacher.errors.blank?
     # Pass in the Shrine-serialized uploaded file to the workflow
     w = workflow.new(@attacher, @options)
-    w.start!(trace: trace)
+    job_id = w.start!(trace: trace)
+    { job_id: job_id, file: @attacher.get }
```

Comment:
I'm not terribly familiar with how Shrine works. In the event that something goes wrong, for example ArgumentError, does there need to be a cleanup step?